WordPress/src/main/java/org/wordpress/android/ui/posts/MediaUploadCompletionProcessor.java

```diff
@@ -0,0 +1,371 @@
+package org.wordpress.android.ui.posts;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Document.OutputSettings;
+import org.jsoup.nodes.Element;
+import org.wordpress.android.editor.Utils;
+import org.wordpress.android.ui.posts.MediaUploadCompletionProcessorPatterns.Helpers;
+import org.wordpress.android.util.helpers.MediaFile;
+
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.wordpress.android.ui.posts.MediaUploadCompletionProcessorPatterns.PATTERN_BLOCK;
+import static org.wordpress.android.ui.posts.MediaUploadCompletionProcessorPatterns.PATTERN_GALLERY_LINK_TO;
+
+public class MediaUploadCompletionProcessor {
```

Comment:
On the second look, I think this file is too big and it'll get bigger in time as we add new blocks that require the same kind of processing. Can we maybe extract out processors that belong to different blocks? 

What I mean is instead of having: processImageBlock, processVideoBlock, processMediaTextBlock, processGalleryBlock all here we could do sth like:

```
//in a different file
interface BlockProcessor {
  process(String block)
}

//in a different file
class ImageBlockProcessor implements BlockProcessor {
    process(String block) {
       ....
    }
}

//in a different file
class VideoBlockProcessor implements BlockProcessor {
    process(String block) {
       ....
    }
}
```

Same with MediaTextBlockBlockProcessor, GalleryBlock etc.

We can even go one step further and define a BlockProcessorFactory that will return the needed processor based on the type, like:

```
class BlockProcessorFactory {

 private ImageBlockProcessor imageProcessor;
 private VideoBlockProcessor imageProcessor
 ...
 public BlockProcessor getProcessor(BlockType type) {
    switch blockType {
      case IMAGE:
         return imageProcessor;
     case VIDEO:
        return videoProcessor;
      ....
    }
 }
}

```
and then processBlock becomes:

    private String processBlock(String block) {
        BlockType type = Helpers.detectBlockType(block);
        blockProcessorFactory.getProcessor(type).process(block);
     }

Reply:
Good suggestion! I've updated the PR after extracting the different processors, and implementing the factory pattern.